datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -90,9 +109,11 @@ struct JoinLeftData {
     /// Counter of running probe-threads, potentially
     /// able to update `visited_indices_bitmap`
     probe_threads_counter: AtomicUsize,
-    /// Memory reservation that tracks memory used by `hash_map` hash table
-    /// `batch`. Cleared on drop.
-    _reservation: MemoryReservation,
```

Comment:
Removing reservation will be sent to the upstream, but upstream does not need `batches_hash_values`, it will be kept in Aras only